onnxruntime/core/optimizer/attention_fusion.cc

```diff
@@ -302,135 +302,19 @@ Status AttentionFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
-      {0, 0, "Add", {7}, kOnnxDomain},
-      {0, 0, "MatMul", {1, 9}, kOnnxDomain},
-      {0, 0, "Reshape", {5}, kOnnxDomain},
-      {0, 0, "Transpose", {1}, kOnnxDomain},
-      {0, 0, "MatMul", {1, 9}, kOnnxDomain},
-      {0, 1, "Transpose", {1}, kOnnxDomain},
-      {0, 0, "Reshape", {5}, kOnnxDomain},
-      {0, 0, "Add", {7}, kOnnxDomain},
-      {0, 0, "MatMul", {1, 9}, kOnnxDomain},
-      {0, 0, "LayerNormalization", {1}, kOnnxDomain}};
-
-  std::vector<const Node::EdgeEnd*> edges;
-  if (!graph_utils::FindPath(add_after_layer_norm, true, parent_path, edges, logger)) {
-    DEBUG_LOG("Faild to find path v");
-    return false;
-  }
-
-  const Node& add = edges[0]->GetNode();
-  const Node& matmul = edges[1]->GetNode();
-  const Node& reshape = edges[2]->GetNode();
-  const Node& transpose = edges[3]->GetNode();
-  const Node& qkv_matmul = edges[4]->GetNode();
-  const Node& v_transpose = edges[5]->GetNode();
-  const Node& v_reshape = edges[6]->GetNode();
-  const Node& v_add = edges[7]->GetNode();
-  const Node& v_matmul = edges[8]->GetNode();
-  const Node& v_root = edges[9]->GetNode();
-  if (v_root.Index() != layer_norm.Index()) {
-    return false;
-  }
-
-  if (!optimizer_utils::CheckOutputEdges(graph, v_add, 1) ||
-      !optimizer_utils::CheckOutputEdges(graph, v_matmul, 1)) {
-    DEBUG_LOG("Output edge count not expected for Add or MatMul in path v");
-    return false;
-  }
-
-  int64_t num_heads = 0;  // will be updated in CheckNodesInPathV
-  int64_t head_size = 0;  // will be updated in CheckNodesInPathV
-  if (!AttentionFusionHelper::CheckNodesInPathV(graph, reshape, transpose, qkv_matmul, v_transpose, v_reshape, num_heads, head_size, hidden_size, logger)) {
-    DEBUG_LOG("CheckNodesInPathV return false");
-    return false;
-  }
-
-  // Validate the input shape of MatMul and Add according to hidden_size.
-  if (!(ValidateAddBiasInitializer(graph, add, hidden_size) &&
-        ValidateMatMulInitializer(graph, matmul, hidden_size) &&
-        ValidateAddBiasInitializer(graph, v_add, hidden_size) &&
-        ValidateMatMulInitializer(graph, v_matmul, hidden_size))) {
-    DEBUG_LOG("Failed in match v_matmul and v_add input shape");
-    return false;
-  }
-
-  // Find mask nodes: Unsqueeze -> Unsqueeze -> (Cast) -> Sub -> Mul -> Add -> Softmax --> [MatMul]
-  // The "Cast" node in parentheses is optional.
-  AttentionFusionHelper::AttentionMaskNodes mask_nodes;
-  if (!AttentionFusionHelper::MatchInputMaskSubgraph(graph, qkv_matmul, mask_nodes, logger, false)) {
-    DEBUG_LOG("Failed in match input mask subgraph");
-    return false;
-  }
-
+static bool FuseSubGraphQKImpl(Node& layer_norm, Graph& graph, AttentionFusionHelper::AttentionMaskNodes& mask_nodes, const std::vector<std::reference_wrapper<const Node>>& parent_path_nodes,
+                                         int64_t hidden_size,  int64_t num_heads, int64_t head_size, std::map<std::string, NodeArg*>& mask_index_map, const logging::Logger& logger) {
```

Comment:
Coding style: align parameters. You can format your code in visual studio 2019 (Ctrl+K, Ctrl+D).